core/src/main/java/dk/alexandra/fresco/framework/sce/resources/storage/FilebasedStreamedStorageImpl.java

```diff
@@ -27,85 +26,72 @@
   /**
    * Creates an instance of the file based streamed storage. For non-streamable object, the given
    * internal storage is used.
-   * 
+   *
    * @param internalStorage The storage used for non-streamable objects
    */
   public FilebasedStreamedStorageImpl(Storage internalStorage) {
+    this(internalStorage, new ConcurrentHashMap<>(), new ConcurrentHashMap<>());
+  }
+
+  /**
+   * Creates an instance of the file based streamed storage. For non-streamable object, the given
+   * internal storage is used.
+   *
+   * @param internalStorage The storage used for non-streamable objects
+   * @param inputs a map from store names to input streams reading from the named store
+   * @param inputs a map from store names to output streams writing to the named store
+   */
+  protected FilebasedStreamedStorageImpl(Storage internalStorage,
+      ConcurrentHashMap<String, ObjectInputStream> inputs,
+      ConcurrentHashMap<String, ObjectOutputStream> outputs) {
```

Comment:
Why concurrent - that seems steep?

Reply:
We used to have multiple threads access these maps via the FiledBasedStreamedStorage. With the current evaluators, I suppose they no longer need to be concurrent.

Reply:
I agree, however synchronous access to a storage also implies a Serialization transaction level on the storage, which in turn renders the synchronous bit irrelevant. My best estimate is to make the resource pool and in turn the storage "splittable"

Reply:
So, for now I am thinking, just simple Map's. In the longer run the whole Storage package probably needs a workover.